core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DedupAggregator.java

```diff
@@ -186,19 +187,35 @@ private void validateSchema(Schema inputSchema, List<String> uniqueFields,
       Schema.Field field = inputSchema.getField(uniqueField);
       if (field == null) {
         collector.addFailure(String.format("Field '%s' does not exist in the input schema", uniqueField), "")
-          .withConfigElement("uniqueFields", uniqueField);
+                .withConfigElement("uniqueFields", uniqueField);
       }
     }
 
     if (function != null) {
       Schema.Field field = inputSchema.getField(function.getField());
       if (field == null) {
         collector.addFailure(String.format("Invalid filter %s(%s): Field '%s' does not exist in input schema ",
-                                           function.getFunction(), function.getField(), function.getField()),
-                             null)
-          .withConfigProperty("filterOperation");
+                                function.getFunction(), function.getField(), function.getField()),
+                        null)
+                .withConfigProperty("filterOperation");
       }
-    }
+      List<Schema.Type> numericSchemaTypes = new ArrayList(Arrays.asList(Schema.Type.INT, Schema.Type.LONG,
+              Schema.Type.FLOAT, Schema.Type.DOUBLE));
+      Schema.Type schemaType;
+      if ((field.getSchema().getType()).equals(Schema.Type.UNION)) {
```

Comment:
Are you checking Unions to check for nullable fields?

If so, here's the correct way to do this:

```
Schema.Type fieldType = isNullable ? field.getSchema().getNonNullable().getType() : field.getSchema().getType();
```

Reply:
got it. addressed this in the next rev